cmd/release-controller/round_robin.go

```diff
@@ -0,0 +1,46 @@
+package main
+
+import (
+	"errors"
+	"sync"
+)
+
+var ErrEmptyItems = errors.New("cannot create round robin with no items")
+
+type RoundRobin struct {
+	current int
+	pool []string
+
+	sync.Mutex
+}
+
+func NewRoundRobin(items...string) (*RoundRobin, error) {
+	if len(items) == 0 {
+		return nil, ErrEmptyItems
+	}
+	return &RoundRobin{
+		pool: items,
+	}, nil
+}
+
+func (r *RoundRobin) Get() string {
+	r.Lock()
+	defer r.Unlock()
+
+	if r.current >= len(r.pool) {
+		r.current = r.current % len(r.pool)
```

Comment:
nit: instead of a mod, if the size is longer than array len, subtract len(array).  This avoids the division.

Reply:
Updated